deepchem/models/torch_models/ferminet.py

```diff
@@ -0,0 +1,132 @@
+from deepchem.utils.electron_sampler import ElectronSampler
+
+# TODO look for the loss function(Hamiltonian)
+
+
+class Ferminet:
+  """A deep-learning based Variational Monte Carlo method for calculating the ab-initio
+    solution of a many-electron system.
+
+    This model aims to calculate the ground state energy of a multi-electron system
+    using a baseline solution as the Hartree-Fock. An MCMC technique is used to sample
+    electrons and DNNs are used to caluclate the square magnitude of the wavefunction,
+    in which electron-electron repulsions also are included in the calculation(in the
+    form of Jastrow factor envelopes). The model requires only the nucleus' coordinates
+    as input.
+
+    This method is based on the following paper:
+
+    Spencer, James S., et al. Better, Faster Fermionic Neural Networks. arXiv:2011.07125,
+    arXiv, 13 Nov. 2020. arXiv.org, http://arxiv.org/abs/2011.07125.
+    """
+
+  def __init__(self,
+               nucleon_coordinates: List[List],
+               seed_no: Optional[int] = None,
+               batch_number: int = 10):
+    """
+    Parameters:
+    -----------
+    nucleon_coordinates:  List[List]
+      A list containing nucleon coordinates as the values with the keys as the element's symbol.
+    seed_no: int, optional (default None)
+      Random seed to use for electron initialization.
+    batch_number: int, optional (default 10)
+      Number of batches of the electron's positions to be initialized.
+
+    """
+    # super(Ferminet, self).__init__()
+
+    self.nucleon_coordinates = nucleon_coordinates
+    self.seed_no = seed_no
+    self.batch_number = batch_number
+
+  def test_f(x: np.ndarray) -> np.ndarray:
+    # dummy function which can be passed as the parameter f. f gives the log probability
+    # TODO replace this function with forward pass of the model in future
+    return 2 * np.log(np.random.uniform(low=0, high=1.0, size=np.shape(x)[0]))
+
+  def prepare_input_stream(self,) -> np.ndarray:
+    """Prepares the one-electron and two-electron input stream for the model.
+
+    Returns:
+    --------
+    one_electron_up: numpy.ndarray
+      numpy array containing one-electron coordinates and distances for the up spin electrons.
+    one_electron_down: numpy.ndarray
+      numpy array containing one-electron coordinates and distances for the down spin electrons
+    two_electron_up: numpy.ndarray
+      numpy array containing two-electron coordinates and distances for the up spin electrons
+    two_electron_down: numpy.ndarray
+      numpy array containing two-electron coordinates and distances for the down spin electrons
+    """
+
+    no_electrons = []
+    nucleons = []
+    self.charge: List = []
+
+    for i in self.nucleon_coordinates:
+      mol = Chem.MolFromSmiles('[' + i[0] + ']')
+      for j in mol.GetAtoms():
+        self.charge.append(j.GetAtomicNum())
+        no_electrons.append([j.GetAtomicNum() - j.GetFormalCharge()])
```

Comment:
This is a very expensive way to look up the atomic number for a symbol.  A much cheaper way is

```python
table = Chem.GetPeriodicTable()
...
num = table.GetAtomicNumber(i[0])
```

The formal charge will always be 0, because you don't specify a formal charge.  But should there be an argument somewhere for the user to specify the total charge of the system?

Reply:
Have changed it now. and for specifying the formal charge, I have allowed the user to input it with the element string itself, like 'H1+' or 'H1-', where the end of the string defines the charge type and the number before it defines the magnitude. Is this fine?

Reply:
Usually the way quantum chemistry programs handle this is that you specify the elements, the total charge, and the spin multiplicity.  See for example https://psicode.org/psi4manual/master/tutorial.html.  By specifying formal charges, you imply that you know where all the electrons are going to end up.  But generally you don't know, and it's the program's job to figure out where they'll go.

Reply:
Since at the start we sample electrons around each atom, in case of a formal charge, an atom is randomly selected for the charge?(I am unsure on how the program will figure out at the start)